forem/forem
# Persist selectedTag filtering of the reading list

## What type of PR is this? (check all applicable)



## Description

This change to Persist selectedTag filtering of the reading list when coming back after reading an article. This is achieved via a query param. Since the project does not use react-router-dom, the selected tag is retrieved via 2 lines of vanilla JS.


## Related Tickets & Documents

Discussion: https://github.com/forem/forem/discussions/20585


## QA Instructions, Screenshots, Recordings

1. Go to the reading list
1. Filter by a tag -- the `selectedTag` query parameter will be added to the URL
1. Refresh or open an article and then press back -- in all scenarios the list of articles will persist the filtering by selected tag
1. Press All Tags -- `selectedTag` query parameter should disappear, all articles should be listed

![image](https://github.com/forem/forem/assets/68788448/52bfb14a-e508-40a6-8af2-4324c7dd9b83)


### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._


## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ X ] I need help with writing tests
